st2common/st2common/services/workflows.py

```diff
@@ -388,12 +386,13 @@ def request_cancellation(ac_ex_db):
     root_ac_ex_db = ac_svc.get_root_execution(ac_ex_db)
 
     if root_ac_ex_db != ac_ex_db and root_ac_ex_db.status not in ac_const.LIVEACTION_CANCEL_STATES:
-        LOG.info('[%s] Cascading cancelation request to parent workflow.', wf_ac_ex_id)
+        LOG.info('[%s] Cascading cancellation request to parent workflow.', wf_ac_ex_id)
+        conductor.request_workflow_state(states.CANCELED)
```

Comment:
Why do we need to call request_workflow_state here?

Reply:
It is always called before (see line 380) even `ac_svc.request_cancellation` is not called to do the job. So moved line 380 to here only for the case if action cancellation is called.